profitbricks-rest/src/test/java/org/apache/jclouds/profitbricks/rest/features/DataCenterApiLiveTest.java

```diff
@@ -0,0 +1,112 @@
+ * limitations under the License.
+ */
+package org.apache.jclouds.profitbricks.rest.features;
+
+import java.util.List;
+import org.apache.jclouds.profitbricks.rest.domain.DataCenter;
+import org.apache.jclouds.profitbricks.rest.domain.Location;
+import org.apache.jclouds.profitbricks.rest.domain.options.DepthOptions;
+import org.apache.jclouds.profitbricks.rest.internal.BaseProfitBricksLiveTest;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+import org.testng.annotations.Test;
+
+@Test(groups = "live", testName = "DataCenterApiLiveTest")
+public class DataCenterApiLiveTest extends BaseProfitBricksLiveTest {
+   
+   @Test
+   public void testGetList() {
+      List<DataCenter> list = dataCenterApi().list(new DepthOptions().depth(1));
+      assertNotNull(list);
+   }
+   
+   @Test
+   public void testCreate() {
+      DataCenter dataCenter = createDataCenter();
+      
+      assertNotNull(dataCenter);
+      assertNotNull(dataCenter.id());
+      
+      deleteDataCenter(dataCenter.id());
+   }
+   
+   @Test
+   public void testGetDataCenter() {
+      DataCenter dataCenter = createDataCenter();
+      
+      dataCenter = dataCenterApi().getDataCenter(dataCenter.id());
+      
+      assertNotNull(dataCenter);
+      assertEquals(dataCenter.properties().name(), "test-data-center");
+      
+      deleteDataCenter(dataCenter.id());
+   }
+      
+   @Test
+   public void testUpdate() {      
+      DataCenter dataCenter = createDataCenter();
+      
+      dataCenter = dataCenterApi().update(dataCenter.id(), "test-data-center2");
+      
+      assertNotNull(dataCenter);
+      assertEquals(dataCenter.properties().name(), "test-data-center2");
+      
+      deleteDataCenter(dataCenter.id());
+   }
+   
+   @Test
+   public void testDelete() {      
+      DataCenter dataCenter = createDataCenter();      
+      dataCenterApi().delete(dataCenter.id());      
+      dataCenter = getDataCenter(dataCenter.id());      
+      //assertEquals(dataCenter, null);
+   }
+   
+   @Test
+   public void testDepth() {
+      
+      DataCenter dataCenter = createDataCenter();
+      
+      for (int i = 1; i <= 5; ++i) {
+         dataCenter = getDataCenter(dataCenter.id(), new DepthOptions().depth(i));
+         assertNotNull(dataCenter);
+      }
+      
+      deleteDataCenter(dataCenter.id());
+   }
+   
+   private DataCenter createDataCenter() {
+      return dataCenterApi().create("test-data-center", "example description", Location.US_LAS.value());
+   }
+   
+   private DataCenter getDataCenter(String id) {
+      return dataCenterApi().getDataCenter(id);
+   }
+   
+   private DataCenter getDataCenter(String id, DepthOptions options) {
+      return dataCenterApi().getDataCenter(id, options);
+   }
+   
+   private void deleteDataCenter(String id) {
+      dataCenterApi().delete(id);
+   }
+   
+   private DataCenterApi dataCenterApi() {
+      return api.dataCenterApi();
+   }
+   
+}
```

Comment:
The test looks OK. Take into account that tests will take a while when the provider is complete, so you should consider chaining the tests and avoid creating a datacenter for each test if it is an expensive operation.